fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/Table.java

```diff
@@ -137,6 +137,17 @@ public void apply(IDatabaseAdapter target) {
         }
     }
 
+    @Override
+    public void apply(Integer priorVersion, IDatabaseAdapter target) {
+        for (Migration step : preSteps) {
+            step.migrateFrom(priorVersion).stream().forEachOrdered(target::runStatement);
+        }
+        apply(target);
+        for (Migration step : postSteps) {
+            step.migrateFrom(priorVersion).stream().forEachOrdered(target::runStatement);
```

Comment:
if you apply this (say 2 steps)
step 1 alter table add column succeeds
step 2 move content (say these are in pre-steps) 
what happens to the above. 

is it idempotent? is it in a deterministic state?  right now I don't think we could successfully migrate beyond one step one version in a deterministic way. 

I think we have to track the migration steps in VERSION_HISTORY and filter the successful steps.

Reply:
maybe even log out that the step is being skipped.